source/MaterialXTest/ShaderGen.cpp

```diff
@@ -36,7 +36,12 @@ void loadLibraries(const mx::StringVec& libraryNames, const mx::FilePath& search
         for (const std::string& filename : filenames)
         {
             mx::FilePath file = path / filename;
-            mx::readFromXmlFile(doc, file);
+            mx::DocumentPtr libDoc = mx::createDocument();
```

Comment:
Fix up import. Need to set a source URI before import so that imported elements have one to check later.